lib/task_helpers/exports/tags.rb

```diff
@@ -0,0 +1,36 @@
+module TaskHelpers
+  class Exports
+    class Tags
+      # Description attribute of Tag Categories that are not visible in the UI
+      SPECIAL_TAGS = ['Parent Folder Path (VMs & Templates)', 'Parent Folder Path (Hosts & Clusters)', 'User roles'].freeze
+
+      def export(options = {})
+        export_dir = options[:directory]
+
+        tags = if options[:all]
+                 Classification.where('parent_id = ? AND description NOT IN (?)', 0, SPECIAL_TAGS)
+               else
+                 export_tags = []
+                 Classification.where('parent_id = ? AND description NOT IN (?)', 0, SPECIAL_TAGS).each do |cat|
+                   if !cat.default
+                     export_tags << cat
+                   else
+                     cat.entries.each do |entry|
+                       unless entry.default
+                         export_tags << cat
+                         break
```

Comment:
I'm not following why we break out after the first non-default entry...

Reply:
This part is hard to follow and is probably why I usually forget how classifications work.  Classifications containing both categories and entries and both could be default is confusing to me.  As long as we have tests for this, I guess it's ok.

Reply:
The goal with this code is to not export default tag categories unless a tag category contains an entry that is not default.

I break out of the loop once one non-default entry is found because at that point the category and all entries will be exported so there is no need to continue looping through the entries.

The test "exports user tags to a given directory" does cover checking this case, but does not check the contents of the exported file. I'll add an expect to ensure that the exported files contents are as expected.